lib/internal/errors.js

```diff
@@ -0,0 +1,282 @@
+// be a string formatted for use with
+// util.format, or a function.
+
+// Utility function for registering the error codes.
+function E(sym, val) {
+  Error[Symbol.for(sym)] = val;
+}
+
+E('ARGNOTNUM', argNotNum);
+E('ASSERTIONERROR', (msg) => msg);
+E('BUFFERLIST',
+  '"buffer" list argument must contain only Buffers or strings');
+E('BUFFERSWAP16', 'Buffer size must be a multiple of 16-bits');
+E('BUFFERSWAP32', 'Buffer size must be a multiple of 32-bits');
+E('BUFFERTOOLARGE', bufferTooBig);
+E('CALLBACKREQUIRED', 'The required "callback" argument is not a function');
+E('CHANNELCLOSED', 'Channel closed');
+E('CLOSEDBEFOREREPLY', 'Slave closed before reply');
+E('CONSOLEWRITABLE', 'Console expects a writable stream instance');
+E('DNSUNKNOWNDOMAIN', 'Unable to determine the domain name');
+E('ECONNRESET', 'Socket hang up');
+E('FILETOOBIG', fileTooBig);
+E('HEADERSSENT',
+  'Unable to set additional headers after they are already sent');
+E('HTTPINVALIDHEADER', 'The header content contains invalid characters');
+E('HTTPINVALIDPATH', 'Request path contains unescaped characters');
+E('HTTPINVALIDSTATUS', (status) => `Invalid status code: ${status}`);
+E('HTTPINVALIDTOKEN', (val) => `Invalid HTTP token: ${val}`);
+E('HTTPINVALIDTRAILER', 'The trailer content contains invalid characters');
+E('HTTPMETHODINVALID', 'Method must be a valid HTTP token');
+E('HTTPUNSUPPORTEDPROTOCOL', unsupportedProtocol);
+E('INDEXOUTOFRANGE', 'Index out of range');
+E('INVALIDARG', invalidArgument);
+E('INVALIDARGVALUE', invalidArgumentValue);
+E('INVALIDFAMILY', 'IP protocol family must be either 4 or 6');
+E('INVALIDOPT', invalidOption);
+E('INVALIDOPTVALUE', invalidOptionValue);
+E('INVALIDPID', 'Invalid pid');
+E('IPCBADHANDLE', 'Handles of this type cannot be sent');
+E('IPCBUFFERASYNCFORK',
+  (stdio) => `Asynchronous forks do not support Buffer input: ${stdio}`);
+E('IPCDISCONNECTED', 'The IPC channel is already disconnected');
+E('IPCNOSYNCFORK', 'IPC cannot be used with synchronous forks');
+E('IPCONEIPC', 'Child process can have only one IPC pipe');
+E('NOCPUUSAGE', (err) => `Unable to obtain CPU Usage: ${err}`);
+E('NOCRYPTO', 'Node.js is not compiled with openssl crypto support');
+E('NOTIMPLEMENTED', notImplemented);
+E('OUTOFBOUNDSARG', (arg) => `"${arg}" argument is out of bounds`);
+E('PORTRANGE', '"port" argument must be a number >= 0 and < 65536');
+E('PROTOTYPEREQUIRED',
+  'The super constructor to "inherits" must have a prototype');
+E('REPLHISTORYPARSE', (path) => `Could not parse history data in ${path}`);
+E('REQUIREDARG', requiredArgument);
+E('STDERRCLOSE', 'process.stderr cannot be closed');
+E('STDOUTCLOSE', 'process.stdout cannot be closed');
+E('UNKNOWNENCODING', (enc) => `Unknown encoding: ${enc}`);
+E('UNKNOWNSIGNAL', (sig) => `Unknown signal: ${sig}`);
+E('UNKNOWNSTATE', 'Unknown state');
+E('WRITEOUTOFBOUNDS', 'Write out of bounds');
+
+// Utility methods for the messages above
+
+function bufferTooBig(max) {
```

Comment:
why make the code `BUFFERTOOLARGE` but name this `bufferTooBig()` instead of `bufferTooLarge()`?